lib/Generator.rb

```diff
@@ -0,0 +1,26 @@
+class Generator
+  def self.generate_asm(ast)
+    output = "SECTION .text\n"
+    traverse(ast.program, output)
+  end
+
+  def self.traverse(ast, output)
+    return traverse(ast.function, output) if ast.is_a? Program
+    
+    if ast.is_a? Function
+      output << "global _#{ast.name}\n\n"
+      output << "_#{ast.name}:\n"
+      return traverse(ast.return, output)
```

Comment:
move out to a separate method
```suggestion
      return traverse_function(ast, output)
```